### ISSUE TITLE
[BUG] GpuDeltaInvariantCheckerExec: Found an expression that shouldn't be here class org.apache.spark.sql.catalyst.expressions.Alias

### ISSUE THREAD

**Issue by tgravescs** [NVIDIA/spark-rapids#7806]
**Describe the bug**

Customer query hit this:

```
: org.apache.spark.SparkException: Job aborted.
	at org.apache.spark.sql.rapids.GpuFileFormatWriter$.write(GpuFileFormatWriter.scala:267)
	at com.databricks.sql.transaction.tahoe.rapids.shims.GpuOptimisticTransactionBase.$anonfun$writeFiles$5(GpuOptimisticTransactionBase.scala:269)
	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withCustomExecutionEnv$8(SQLExecution.scala:239)
	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:386)
	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withCustomExecutionEnv$1(SQLExecution.scala:186)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:968)
	at org.apache.spark.sql.execution.SQLExecution$.withCustomExecutionEnv(SQLExecution.scala:141)
	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:336)
	at com.databricks.sql.transaction.tahoe.rapids.shims.GpuOptimisticTransactionBase.writeFiles(GpuOptimisticTransactionBase.scala:228)
	at com.databricks.sql.transaction.tahoe.commands.WriteIntoDelta.liftedTree1$1(WriteIntoDelta.scala:321)
	at com.databricks.sql.transaction.tahoe.commands.WriteIntoDelta.write(WriteIntoDelta.scala:321)
	at com.databricks.sql.transaction.tahoe.rapids.shims.GpuWriteIntoDelta.$anonfun$run$1(GpuWriteIntoDelta.scala:44)
	at com.databricks.sql.transaction.tahoe.rapids.shims.GpuWriteIntoDelta.$anonfun$run$1$adapted(GpuWriteIntoDelta.scala:37)
	at com.databricks.sql.transaction.tahoe.rapids.shims.GpuDeltaLog.withNewTransaction(GpuDeltaLog.scala:56)
	at com.databricks.sql.transaction.tahoe.rapids.shims.GpuWriteIntoDelta.run(GpuWriteIntoDelta.scala:37)
	at com.nvidia.spark.rapids.delta.shims.GpuDeltaDataSource.createRelation(GpuDeltaDataSource.scala:57)
	at org.apache.spark.sql.execution.datasources.SaveIntoDataSourceCommand.run(SaveIntoDataSourceCommand.scala:47)
	at com.nvidia.spark.rapids.GpuExecutedCommandExec.sideEffectResult$lzycompute(GpuExecutedCommandExec.scala:51)
	at com.nvidia.spark.rapids.GpuExecutedCommandExec.sideEffectResult(GpuExecutedCommandExec.scala:49)
	at com.nvidia.spark.rapids.GpuExecutedCommandExec.executeCollect(GpuExecutedCommandExec.scala:60)
	at org.apache.spark.sql.execution.QueryExecution$$anonfun$$nestedInanonfun$eagerlyExecuteCommands$1$1.$anonfun$applyOrElse$1(QueryExecution.scala:160)
	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withCustomExecutionEnv$8(SQLExecution.scala:239)
	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:386)
	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withCustomExecutionEnv$1(SQLExecution.scala:186)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:968)
	at org.apache.spark.sql.execution.SQLExecution$.withCustomExecutionEnv(SQLExecution.scala:141)
	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:336)
	at org.apache.spark.sql.execution.QueryExecution$$anonfun$$nestedInanonfun$eagerlyExecuteCommands$1$1.applyOrElse(QueryExecution.scala:160)
	at org.apache.spark.sql.execution.QueryExecution$$anonfun$$nestedInanonfun$eagerlyExecuteCommands$1$1.applyOrElse(QueryExecution.scala:156)
	at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$transformDownWithPruning$1(TreeNode.scala:590)
	at org.apache.spark.sql.catalyst.trees.CurrentOrigin$.withOrigin(TreeNode.scala:168)
	at org.apache.spark.sql.catalyst.trees.TreeNode.transformDownWithPruning(TreeNode.scala:590)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.org$apache$spark$sql$catalyst$plans$logical$AnalysisHelper$$super$transformDownWithPruning(LogicalPlan.scala:30)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.transformDownWithPruning(AnalysisHelper.scala:268)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.transformDownWithPruning$(AnalysisHelper.scala:264)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.transformDownWithPruning(LogicalPlan.scala:30)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.transformDownWithPruning(LogicalPlan.scala:30)
	at org.apache.spark.sql.catalyst.trees.TreeNode.transformDown(TreeNode.scala:566)
	at org.apache.spark.sql.execution.QueryExecution.$anonfun$eagerlyExecuteCommands$1(QueryExecution.scala:156)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper$.allowInvokingTransformsInAnalyzer(AnalysisHelper.scala:324)
	at org.apache.spark.sql.execution.QueryExecution.eagerlyExecuteCommands(QueryExecution.scala:156)
	at org.apache.spark.sql.execution.QueryExecution.commandExecuted$lzycompute(QueryExecution.scala:141)
	at org.apache.spark.sql.execution.QueryExecution.commandExecuted(QueryExecution.scala:132)
	at org.apache.spark.sql.execution.QueryExecution.assertCommandExecuted(QueryExecution.scala:186)
	at org.apache.spark.sql.DataFrameWriter.runCommand(DataFrameWriter.scala:959)
	at org.apache.spark.sql.DataFrameWriter.saveToV1Source(DataFrameWriter.scala:427)
	at org.apache.spark.sql.DataFrameWriter.saveInternal(DataFrameWriter.scala:338)
	at org.apache.spark.sql.DataFrameWriter.save(DataFrameWriter.scala:250)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:380)
	at py4j.Gateway.invoke(Gateway.java:295)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:251)
	at java.lang.Thread.run(Thread.java:750)
Caused by: java.lang.IllegalArgumentException: Found an expression that shouldn't be here class org.apache.spark.sql.catalyst.expressions.Alias
	at com.nvidia.spark.rapids.GpuBindReferences$.$anonfun$postBindCheck$1(GpuBoundAttribute.scala:54)
	at com.nvidia.spark.rapids.GpuBindReferences$.$anonfun$postBindCheck$1$adapted(GpuBoundAttribute.scala:45)
	at org.apache.spark.sql.catalyst.trees.TreeNode.foreach(TreeNode.scala:363)
	at com.nvidia.spark.rapids.GpuBindReferences$.postBindCheck(GpuBoundAttribute.scala:45)
	at com.nvidia.spark.rapids.GpuBindReferences$.bindRefInternal(GpuBoundAttribute.scala:94)
	at com.nvidia.spark.rapids.GpuBindReferences$.bindGpuReference(GpuBoundAttribute.scala:101)
	at com.nvidia.spark.rapids.GpuBindReferences$.$anonfun$bindGpuReferences$1(GpuBoundAttribute.scala:111)
	at scala.collection.immutable.List.map(List.scala:297)
	at com.nvidia.spark.rapids.GpuBindReferences$.bindGpuReferences(GpuBoundAttribute.scala:111)
	at com.nvidia.spark.rapids.GpuProjectExec.doExecuteColumnar(basicPhysicalOperators.scala:154)
	at org.apache.spark.sql.execution.SparkPlan.$anonfun$executeColumnar$1(SparkPlan.scala:253)
	at org.apache.spark.sql.execution.SparkPlan.$anonfun$executeQuery$1(SparkPlan.scala:270)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:165)
	at org.apache.spark.sql.execution.SparkPlan.executeQuery(SparkPlan.scala:266)
	at org.apache.spark.sql.execution.SparkPlan.executeColumnar(SparkPlan.scala:249)
	at com.databricks.sql.transaction.tahoe.rapids.shims.GpuDeltaInvariantCheckerExec.doExecuteColumnar(GpuDeltaInvariantCheckerExec.scala:48)
	at org.apache.spark.sql.execution.SparkPlan.$anonfun$executeColumnar$1(SparkPlan.scala:253)
	at org.apache.spark.sql.execution.SparkPlan.$anonfun$executeQuery$1(SparkPlan.scala:270)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:165)
	at org.apache.spark.sql.execution.SparkPlan.executeQuery(SparkPlan.scala:266)
	at org.apache.spark.sql.execution.SparkPlan.executeColumnar(SparkPlan.scala:249)
	at com.nvidia.spark.rapids.GpuSortExec.doExecuteColumnar(GpuSortExec.scala:142)

```

**Comment by jlowe** [comment]
I believe this is already fixed by #7455.  @tgravescs could you try running with a version that has that fix?

**Comment by jlowe** [comment]
Verified with @tgravescs that this can be closed.
